File: collabbackend1/src/main/java/com/niit/service/FriendService.java
package com.niit.service;

import java.util.List;

import com.niit.model.Friend;
import com.niit.model.User;

public interface FriendService {
	List <User> listOfSuggestedUsers(String username);

	void friendRequest(Friend friend);
	List<Friend> pendingRequests(String toId);

	void updateFriendRequest(Friend friend);
	List<String>listOfFriends(String username);


}

File: collabfrontend1/WebContent/services/friendservice.js
/**
 * 
 */
app.factory('FriendService',function($http){
	var friendService={}
	var BASE_URL="http://localhost:8088/collabmiddleend"
		friendService.listOfSuggestedUsers=function()
		{
		return $http.get(BASE_URL+"/getsuggestedusers")
		}
	friendService.sendFriendRequest=function(toId)
	{
	return $http.get(BASE_URL+"/friendrequest/"+toId)
	}
	
	friendService.pendingRequests=function()
	{
	return $http.get(BASE_URL+"/pendingrequests")
	}
	
	friendService.updatePendingRequest=function(request)
	{
	return $http.put(BASE_URL+"/updatependingrequest",request)
	}
	friendService.listOfFriends=function()
	{
	return $http.get(BASE_URL+"/friendslist")
	}
	return friendService;
})